enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,171 @@
+authors:
+  - "@wking"
+reviewers:
+  - "@abhinavdahiya"
+  - "@crawford"
+approvers:
+  - TBD
+creation-date: 2019-11-19
+last-updated: 2019-11-19
+status: implementable
+---
+
+# Available-update Metadata
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently the ClusterVersion API has [a single `Update` type][api-update] used for both [the `desiredUpdate` input][api-desiredUpdate] and the [`desired`][api-desired] and [`availableUpdates`][api-availableUpdates] outputs.
+But update requests like `desiredUpdate` have different schema requirements than output like `desired` and `availableUpdates`.
+For example, [the `force` property][api-force] makes sense in the `desiredUpdate` context, but not in the `availableUpdates` contexts.
+And release metadata like landing page URIs and release metadata makes sense in the `desired` and `availableUpdates` contexts, but not in the `desiredUpdate` context.
+This enhancement adds a new `Release` type to be used in `status` properties so we can distinguish between input and output schema.
+
+## Motivation
+
+Cincinnati provides [metadata about releases][cincinnati-metadata], including (since [here][cincinnati-channel-metadata]).
+Exposing that information to ClusterVersion consumers allows them to use live metadata, instead of talking to Cincinnati directly or hard-coding expected values.
+For example, the console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.metadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], to add a new type:
+
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
```

Comment:
Update vs Release not sure the names make ton of sense..

Reply:
> Update vs Release not sure the names make ton of sense

Updates are transitions (edges) between releases (nodes).